### PR TITLE
Added support for versionAliases field in google_secret_manager_secret resource

### PR DESCRIPTION
**Pull request by abheda-crest**
Added support for `version_aliases` field in `google_secret_manager_secret` resource.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
secretmanager: added `version_aliases` field to `google_secret_manager_secret` resource
```

### CODE REVIEW

**Review by NickElliot — COMMENTED** [review]
Looks good, but I want to clarify: is there a scenario you're trying to test for with the 5-step update test? The situation I assume of "you can add aliases and remove them" could be covered by a simpler 3-step test like there is for [annotations.](https://github.com/GoogleCloudPlatform/magic-modules/blob/c67a67923b62c07ffd8421eebb69369621870d5a/mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb#L67)

**Comment by abheda-crest** [comment]
> Looks good, but I want to clarify: is there a scenario you're trying to test for with the 5-step update test? The situation I assume of "you can add aliases and remove them" could be covered by a simpler 3-step test like there is for [annotations.](https://github.com/GoogleCloudPlatform/magic-modules/blob/c67a67923b62c07ffd8421eebb69369621870d5a/mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb#L67)

Unlike the `annotations` field, we cannot provide the versionAliases field while creating the secret because the version alias can only be mapped to existing versions. Hence, the first step creates a secret without this field and multiple secret versions. The second step adds this field to the resource while the subsequent steps check various update scenarios such as updating keys, updating values, updating key-value pair, adding key-value pair, and removing key-value pair. The last step removes the entire `version_aliases` field from the resource.

File: mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb [hunk 1]
```diff
@@ -107,6 +107,67 @@ func TestAccSecretManagerSecret_annotationsUpdate(t *testing.T) {
...
+				Config: testAccSecretManagerSecret_versionAliasesUpdate(context),
+			},
+			{
+				ResourceName:      "google_secret_manager_secret.secret-basic",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{"ttl"},
+			},
+			{
+				Config: testAccSecretManagerSecret_versionAliasesBasic(context),
```
Comment (NickElliot) [inline mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb:150]:
```suggestion
```

**Review by NickElliot — CHANGES_REQUESTED** [review]
I see, thank you for the explanation! In that case, would it be possible to remove just the 4th step, as repeating the testAccSecretManagerSecret_versionAliasesBasic version is not testing any functionality not tested via the original Basic-Updated version? I.e. step 3 is doing a key update, value update, removal, and addition, while step 4 repeats the same four operation tests.

Because of how TPG creates API requests, the ordering of the fields in a list is irrelevant (they all get sent in alphabetical order of the keys), so there is not a scenario being tested via adding a KVP in the middle of the list vs the end (which appears to be the only distinction between the basic->updated and updated->basic scenarios).

File: mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb (hunk 1 above)
Comment (abheda-crest) [reply mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb]: Okay, I've removed step 4 of this test.

**Comment by abheda-crest** [comment]
@NickElliot Could you please start the tests as it is awaiting your approval and merge this PR?

**Review by NickElliot — APPROVED** [review]
LGTM!

**Comment by NickElliot** [comment]
However it needs the merge conflict resolved now, sorry for the delay that caused that

**Comment by abheda-crest** [comment]
> However it needs the merge conflict resolved now, sorry for the delay that caused that

No problem, I've resolved the merge conflict. Could you please trigger the tests again and merge this PR?

**Comment by henry-chew-partior** [comment]
> > Looks good, but I want to clarify: is there a scenario you're trying to test for with the 5-step update test? The situation I assume of "you can add aliases and remove them" could be covered by a simpler 3-step test like there is for [annotations.](https://github.com/GoogleCloudPlatform/magic-modules/blob/c67a67923b62c07ffd8421eebb69369621870d5a/mmv1/third_party/terraform/tests/resource_secret_manager_secret_test.go.erb#L67)
> 
> Unlike the `annotations` field, we cannot provide the versionAliases field while creating the secret because the version alias can only be mapped to existing versions. Hence, the first step creates a secret without this field and multiple secret versions. The second step adds this field to the resource while the subsequent steps check various update scenarios such as updating keys, updating values, updating key-value pair, adding key-value pair, and removing key-value pair. The last step removes the entire `version_aliases` field from the resource.

If version_alias can't be set during secrets creation, does it make more sense that the version_alias field should be implemented during creation of the secret version resource then?

**Pull request merged**